### DISCUSSION TITLE
What ui component technology is used behind m365 copilot in teams?

### DISCUSSION THREAD

**Discussion by alexChiu127** [OfficeDev/microsoft-365-agents-toolkit#12597] [Q&A]
Could you please clarify whether the chat UI components in Teams, such as new chat, selecting conversation history, and choosing agents, for M365 Copilot in Teams or its custom declarative copilot, are implemented using web technologies (e.g., Node.js or a JS front-end framework embedded in Teams), or if they can be customized further just using the Bot Framework?

What technology is used behind it?

**Comment by therealjohn** [comment]
It's not customizable or part of any SDK available to developers yet.
